src/Inventory/Asset/MainAsset.php

```diff
@@ -710,7 +710,15 @@ public function rulepassed($items_id, $itemtype, $rules_id, $ports_id = 0)
         //handle domains
         if (property_exists($val, 'domains_id')) {
             $domain = new \Domain();
-            if (!$domain->getFromDBByCrit(Sanitizer::sanitize(['name' => $val->domains_id]))) {
+            if (
+                !$domain->getFromDBByCrit(Sanitizer::sanitize(
+                    [
+                        'name' => $val->domains_id,
+                        'is_deleted' => 0,
+                        'entities_id' => $entities_id,
```

Comment:
Using just `$entitied_id` won't take care of entites inheritance as far as I understand; but it should should'nt it?

Reply:
I have a doubt, using inheritance could return several results .... =/

Reply:
I'm almost always lost talking about inheritance :/ Anyways, change proposal will at least fix some cases already, so OK for me.

Reply:
There is no unicity on `name`+`entities_id`, so using `getFromDBByCrit` could anyway be a problem here.
If we found multiple entries using the entities visibility criteria, it means that there are duplicated entries in the GLPI and it is not the inventory responsibility to fix this.

The following code will permit to attach asset to a domain that has been created in a parent entity, and will also handle presence of multiple domains with the same name in the current entity scope.

```php
$matching_domains = $DB->request([
    'name' => $val->domains_id,
    'is_deleted' => 0,
    'entities_id' => getEntitiesRestrictCriteria($domain->getTable(), '', $entities_id, true),
]);
if ($matching_domains->count() > 0) {
    // Get the first domain, as we assume that a domain should not be declared multiple times in the same entity scope.
    $domain->getFromResultSet($matching_domains->current());
} else {
    $domain->add(...);
}
```